### PR TITLE
[7.17] ci: configure GCP at the stage level

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Partially backport https://github.com/elastic/beats/pull/30907/commits/4820498241c82690837d53900dbc49e86b3f59d0 to help with the merge conflicts

## Why is it important?

Allow stage level customisation to configure GCP

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -333,29 +333,45 @@ def k8sTest(Map args = [:]) {
...
-        retryWithSleep(retries: 3, seconds: 5, backoff: true){
-          try {
-            sh(label: "Setup kind", script: ".ci/scripts/kind-setup.sh")
-          } catch(err) {
-            i++
-            sh(label: 'Delete cluster', script: 'kind delete cluster')
-            if (i > 2) {
-              error("Setup kind failed with error '${err.toString()}'")
-            }
+    withK8s() {
```
Comment (kuisathaverat) [inline Jenkinsfile:336]: 🤔 if `k8s` is **false** then run with k8s. It is a little odd to me, the other way around makes more sense to me.

File: Jenkinsfile [hunk 2]
```diff
@@ -333,29 +333,45 @@ def k8sTest(Map args = [:]) {
...
+      body()
+    }
+  } else if (args.get('gcp', false)) {
```
Comment (kuisathaverat) [inline Jenkinsfile:339]: ditto

File: Jenkinsfile (hunk 1 above)
Comment (v1v) [reply Jenkinsfile:336]: that's not the case, it's the default value `map.get('value', otherwise-default-value)`

Comment (v1v) [reply Jenkinsfile:336]: I could have done:

```
def k8s = args.containsKey('k8s') ? args.k8s  : false
if (k8s)

```

but I used `if (args.get('k8s', false)) {` instead

Comment (kuisathaverat) [reply Jenkinsfile:336]: 🤦‍♂️

**Pull request merged**